src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/ppc64/aix/AixPPC64VaList.java

```diff
@@ -0,0 +1,441 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * ===========================================================================
+ * (c) Copyright IBM Corp. 2021, 2021 All Rights Reserved
+ * ===========================================================================
+ */
+
+package jdk.internal.foreign.abi.ppc64.aix;
+
+import jdk.incubator.foreign.*;
+import jdk.internal.foreign.ResourceScopeImpl;
+import jdk.internal.foreign.Scoped;
+import jdk.internal.foreign.Utils;
+import jdk.internal.foreign.abi.SharedUtils;
+import jdk.internal.misc.Unsafe;
+
+import java.lang.invoke.VarHandle;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static jdk.internal.foreign.PlatformLayouts.AIX;
+
+import static jdk.incubator.foreign.MemoryLayout.PathElement.groupElement;
+import static jdk.internal.foreign.abi.SharedUtils.SimpleVaArg;
+import static jdk.internal.foreign.abi.SharedUtils.THROWING_ALLOCATOR;
+
+/**
+ * This file is copied from x86/sysv (Linux/x86_64) as a placeholder for compilation as VaList
+ * on AIX/ppc64le at Java level is not yet implemented for the moment. The definition of VaList
+ * corresponds to the underlying struct of va_list defined on AIX/ppc64le which appears to be a
+ * 4-byte pointer to the actual argument value in debugging (totaly distinct from the existing
+ * struct defined in Windows or Linux on x86_64). Thus, futher analysis on the struct is required
+ * to understand how the struct is laid out in memory according to the description in the publisized
+ * ABI document at https://refspecs.linuxfoundation.org/ELF/ppc64/PPC-elf64abi-1.9.pdf.
+ */
+public non-sealed class AixPPC64VaList implements VaList, Scoped {
+    private static final Unsafe U = Unsafe.getUnsafe();
+
+    static final Class<?> CARRIER = MemoryAddress.class;
+
+//    struct typedef __va_list_tag __va_list_tag {
+//        unsigned int               gp_offset;            /*     0     4 */
+//        unsigned int               fp_offset;            /*     4     4 */
+//        void *                     overflow_arg_area;    /*     8     8 */
+//        void *                     reg_save_area;        /*    16     8 */
+//
+//        /* size: 24, cachelines: 1, members: 4 */
+//        /* last cacheline: 24 bytes */
+//    };
+    static final GroupLayout LAYOUT = MemoryLayout.structLayout(
+        AIX.C_INT.withName("gp_offset"),
+        AIX.C_INT.withName("fp_offset"),
+        AIX.C_POINTER.withName("overflow_arg_area"),
+        AIX.C_POINTER.withName("reg_save_area")
+    ).withName("__va_list_tag");
+
+    private static final MemoryLayout GP_REG = MemoryLayout.paddingLayout(64).withBitAlignment(64);
+    private static final MemoryLayout FP_REG = MemoryLayout.paddingLayout(128).withBitAlignment(128);
+
+    private static final GroupLayout LAYOUT_REG_SAVE_AREA = MemoryLayout.structLayout(
```

Comment:
these are x86-64 registers, why are they here?

Reply:
Is this based on the x86 version of AixPPC64VaList.java?

Reply:
As I mentioned in the comments of these code, all Valist related code (including AIX/PPC64, Linux/PPC64LE and Linux/s390x) are placeholder to have compilation passed (we never use for the moment), which will be totally rewritten in the future after we figure out how to modelling the Valist on these platforms.

Reply:
The Valist related files include:
```
jdk/internal/foreign/abi/ppc64/aix/AixPPC64VaList.java
jdk/internal/foreign/abi/ppc64/aix/ArgumentClassImpl.java
jdk/internal/foreign/abi/ppc64/aix/TypeClass.java

jdk/internal/foreign/abi/ppc64/sysv/SysVPPC64leVaList.java
jdk/internal/foreign/abi/ppc64/sysv/TypeClass.java

jdk/internal/foreign/abi/s390x/sysv/ArgumentClassImpl.java
jdk/internal/foreign/abi/s390x/sysv/SysVS390xVaList.java
jdk/internal/foreign/abi/s390x/sysv/TypeClass.java
```

Reply:
We already added these stubs for the public APIs in there by throwing the exception if it is intended to operate on Valist.
e.g. 
jdk/internal/foreign/abi/ppc64/aix/AixPPC64VaList.java

```
    @Override
    public int nextVarg(ValueLayout.OfInt layout) {
        throw new InternalError("nextVarg is not yet implemented"); //$NON-NLS-1$
    }
...
    @Override
    public void skip(MemoryLayout... layouts) {
        throw new InternalError("skip is not yet implemented"); //$NON-NLS-1$
    }

    static AixPPC64VaList.Builder builder(ResourceScope scope) {
        return new AixPPC64VaList.Builder(scope);
    }

    public static VaList ofAddress(MemoryAddress ma, ResourceScope scope) {
        throw new InternalError("ofAddress is not yet implemented"); //$NON-NLS-1$
    }
...
        @Override
        public Builder addVarg(ValueLayout.OfInt layout, int value) {
            throw new InternalError("addVarg is not yet implemented"); //$NON-NLS-1$
        }
```